### PR TITLE
caching for properties of ideals

### PR DESCRIPTION
**Pull request by simonbrandhorst**
We cache for instance is an ideal is prime. But also the radical idea.
@fingolfin  how can I make `@attr function radical(I::MPolyIdeal)` type stable?

### CODE REVIEW

**Comment by thofma** [comment]
`@atttr T radical(I::T) where {T <: MPolyIdeal}` should work

**Comment by simonbrandhorst** [comment]
Thanks works.

File: src/Rings/mpoly-ideals.jl [hunk 1]
```diff
@@ -569,7 +572,7 @@ julia> minpoly(a)
 x^2 + 1
 ```
 """
-@attr function absolute_primary_decomposition(I::MPolyIdeal{<:MPolyRingElem{QQFieldElem}})
+@attr function absolute_primary_decomposition(I::T) where {T <: MPolyIdeal{<:MPolyRingElem{QQFieldElem}}}
```
Comment (thofma) [inline src/Rings/mpoly-ideals.jl]: `T` not used yet.

Comment (simonbrandhorst) [reply src/Rings/mpoly-ideals.jl]: Removed the `T`. But it is not type stable anyways.

**Review by wdecker — APPROVED** [review]

**Pull request merged**